### DISCUSSION TITLE
[TECH-DEBT] Formulario se corta por mensaje enviado por su trigger

### DISCUSSION THREAD

**Discussion by seiturra** [Cotalker/documentation#274] [4. Bug Reports] [answered]
Buenas,

Tengo una implementación donde envío un formulario con dos survey+survey y su trigger lo único que hace es cambiar de estado + enviar un mensaje al canal. En el caso donde ocurrió el error agregué un sub-survey a cada question survey+survey (o sea, había dos sub surveys) y al enviar el formulario, el mensaje que envió el trigger quedó entre medio del formulario, reemplazando al segundo sub survey (no se muestra el segundo sub survey).
Dejo evidencia del ejemplo.
![image](https://user-images.githubusercontent.com/12589784/122138805-7bb49780-ce15-11eb-8314-078a64a1eaaf.png)

Ambos fragmentos del formulario pasaron a modo de edición independientemente usando la interfaz: click derecho -> editar formulario. No llegaron al canal así ya que fue levantado usando el botón verde. 

Este comportamiento errático no sucede siempre que se ejecuta esta rutina.

Saludos.

**Comment by NicolasPLG** [comment]
@SeIturra puedes probar agregando la rutina el esperar / wait antes de usar el pbmessage, asi te aseguras que llega el survey no cortado.

![image](https://user-images.githubusercontent.com/61419471/122417390-6fd0ee80-cf57-11eb-81e1-9926f9df57dc.png)

**Reply by MiikeVi** [reply]
@NicolasPLG  Gracias por el workaround. Bug levantado igualmente, gracias @SeIturra

**Comment by guillevalin** [comment, marked as answer]
@SeIturra como te comentó @NicolasPLG esto ocurre porque el mensaje de texto llega "antes" que los formularios. Pongo "antes" entre comillas porque la forma de como funcionan los survey+survey es que se agrega un delay artificial de 1 segundo por cada uno de ellos para poder cargar los campos del subformulario de manera "in-line".

Dado que el mensaje de texto no tiene conocimiento de esto, podría quedar con una fecha de creación entre medio de los mensajes del survey, ocasionando el efecto que se ve en la foto. La forma de asegurarse que no pase es que el delay en segundos sea mayor a la cantidad de preguntas de tipo subformulario, ej: si tengo 10 campos survey+survey, debería agregar un delay de 10 segundos por cautela.
